cpp/src/column/column.cu

```diff
@@ -275,6 +278,93 @@ struct create_column_from_view_vector {
 
 };
 
+struct concatenate_using_partition_map {
+  std::vector<column_view> const& views;
+  cudaStream_t stream;
+  rmm::mr::device_memory_resource* mr;
+
+  template <typename T,
+      std::enable_if_t<is_fixed_width<T>()>* = nullptr>
+  std::unique_ptr<column> operator()() {
+
+    // Compute the partition offsets
+    thrust::host_vector<size_type> offsets(views.size() + 1);
+    thrust::transform_inclusive_scan(thrust::host,
+      views.begin(), views.end(), offsets.begin() + 1,
```

Comment:
This is largely a personal preference thing, so if you prefer the `+1` then leave it that way. 
```suggestion
      views.begin(), views.end(), std::next(offsets.begin()),
```

Reply:
Ok, I'll try it out